txp/Pos/Txp/MemState/Class.hs

```diff
@@ -28,21 +28,22 @@ import           Control.Monad.Trans.Control (MonadBaseControl)
 import           Data.Default                (Default (def))
 import qualified Data.HashMap.Strict         as HM
 import           Ether.Internal              (HasLens (..))
+import           Mockable                    (CurrentTime, Mockable, currentTime)
 import           System.IO.Unsafe            (unsafePerformIO)
 import           System.Wlog                 (WithLogger, getLoggerName, usingLoggerName)
 
 import           Pos.Txp.Core.Types          (TxAux, TxId, TxOutAux)
 import           Pos.Txp.MemState.Types      (GenericTxpLocalData (..),
                                               GenericTxpLocalDataPure, TxpMetrics (..))
 import           Pos.Txp.Toil.Types          (MemPool (..), UtxoModifier)
-import           Pos.Util.TimeWarp           (currentTime)
 
 data TxpHolderTag
 
 -- | More general version of @MonadReader (GenericTxpLocalData mw, TxpMetrics) m@.
 type MonadTxpMem ext ctx m
      = ( MonadReader ctx m
        , HasLens TxpHolderTag ctx (GenericTxpLocalData ext, TxpMetrics)
+       , Mockable CurrentTime m
```

Comment:
Personally I don't like it here. When I see `MonadBlaMem` I treat it as an extensible version of `MonadReader BlaContext`.